dspace-spring-rest/src/main/java/org/dspace/app/rest/link/relation/EntityTypeHalLinkFactory.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.link.relation;
+
+import java.util.LinkedList;
+
+import org.dspace.app.rest.RelationshipTypeRestController;
+import org.dspace.app.rest.link.HalLinkFactory;
+import org.dspace.app.rest.model.hateoas.EntityTypeResource;
+import org.springframework.data.domain.Pageable;
+import org.springframework.hateoas.Link;
+import org.springframework.stereotype.Component;
+
+@Component
+public class EntityTypeHalLinkFactory extends HalLinkFactory<EntityTypeResource, RelationshipTypeRestController> {
```

Comment:
Javadocs to describe the purpose of this class

Reply:
The JavaDocs have been included now